ipfs-migrator: 6 -> 7

xeji opened the pull request:
###### Motivation for this change

New version of the ipfs repo migration tools. This is required to migrate existing ipfs repos to repo v7.

Needs backport to 18.09 since existing ipfs repos from 18.03 will have to be migrated to v7 in 18.09.

###### Things done

- [x] built in a sandbox on NixOS

---

xeji committed:
ipfs-migrator: 6 -> 7

New version is required to migrate existing ipfs repos to repo v7.

xeji commented:
@GrahamcOfBorg build ipfs-migrator

GrahamcOfBorg commented:
<!--REQUEST_ID=244636b2-3a59-4f0d-b421-8b779e0c8278-->
Failure on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.46747&attempt_id=4da19108-f779-4c46-8077-4d3a0ad3a6b5)

Attempted: ipfs-migrator

<details><summary>Partial log (click to expand)</summary><p>

```
github.com/ipfs/fs-repo-migrations/Godeps/_workspace/src/github.com/anacrolix/jitter
github.com/ipfs/fs-repo-migrations/Godeps/_workspace/src/github.com/jbenet/go-multiaddr
github.com/ipfs/fs-repo-migrations/Godeps/_workspace/src/github.com/anacrolix/utp
github.com/ipfs/fs-repo-migrations/ipfs-2-to-3/Godeps/_workspace/src/github.com/whyrusleeping/go-logging
github.com/ipfs/fs-repo-migrations/Godeps/_workspace/src/github.com/jbenet/go-multiaddr-net/utp
github.com/ipfs/fs-repo-migrations/Godeps/_workspace/src/github.com/jbenet/go-multiaddr-net
github.com/ipfs/fs-repo-migrations/sharness/dependencies/pollEndpoint
github.com/ipfs/fs-repo-migrations/stump
builder for '/nix/store/3v433gb5pqhslnn7pp9w1w8bdkkkncg1-ipfs-migrator-1.4.0.drv' failed with exit code 3
error: build of '/nix/store/3v433gb5pqhslnn7pp9w1w8bdkkkncg1-ipfs-migrator-1.4.0.drv' failed
```
</p></details>

GrahamcOfBorg commented:
<!--REQUEST_ID=244636b2-3a59-4f0d-b421-8b779e0c8278-->
Success on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.46747&attempt_id=a511ef89-f04c-43cd-a2cf-9b768544f4ee)

Attempted: ipfs-migrator

<details><summary>Partial log (click to expand)</summary><p>

```
stripping (with command strip and flags -S) in /nix/store/k1qjrp714ahnq1zs2p8byngp2dgm2xbm-ipfs-migrator-1.4.0-bin/bin
patching script interpreter paths in /nix/store/k1qjrp714ahnq1zs2p8byngp2dgm2xbm-ipfs-migrator-1.4.0-bin
checking for references to /build in /nix/store/k1qjrp714ahnq1zs2p8byngp2dgm2xbm-ipfs-migrator-1.4.0-bin...
cannot find section .dynamic
cannot find section .dynamic
cannot find section .dynamic
cannot find section .dynamic
cannot find section .dynamic
strip is /nix/store/y4ymnvgxygpq05h03kyzbj572zmh6zla-binutils-2.30/bin/strip
/nix/store/k1qjrp714ahnq1zs2p8byngp2dgm2xbm-ipfs-migrator-1.4.0-bin
```
</p></details>

GrahamcOfBorg commented:
<!--REQUEST_ID=244636b2-3a59-4f0d-b421-8b779e0c8278-->
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.46747&attempt_id=5b8f59cd-4cab-4e65-a6b9-0826c45b0224)

Attempted: ipfs-migrator

<details><summary>Partial log (click to expand)</summary><p>

```
stripping (with command strip and flags -S) in /nix/store/fr1wks0qh0811pvi3as5a7zg4yvcn0jw-ipfs-migrator-1.4.0-bin/bin
patching script interpreter paths in /nix/store/fr1wks0qh0811pvi3as5a7zg4yvcn0jw-ipfs-migrator-1.4.0-bin
checking for references to /build in /nix/store/fr1wks0qh0811pvi3as5a7zg4yvcn0jw-ipfs-migrator-1.4.0-bin...
cannot find section .dynamic
cannot find section .dynamic
cannot find section .dynamic
cannot find section .dynamic
cannot find section .dynamic
strip is /nix/store/h0lbngpv6ln56hjj59i6l77vxq25flbz-binutils-2.30/bin/strip
/nix/store/fr1wks0qh0811pvi3as5a7zg4yvcn0jw-ipfs-migrator-1.4.0-bin
```
</p></details>

GrahamcOfBorg commented:
<!--REQUEST_ID=b6152b0d-76c6-4f32-a66d-47e1c1a59891-->
Success on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.46747&attempt_id=892364b6-dfb5-4687-b454-73c5148e0a33)

Attempted: ipfs-migrator

<details><summary>Partial log (click to expand)</summary><p>

```
/nix/store/k1qjrp714ahnq1zs2p8byngp2dgm2xbm-ipfs-migrator-1.4.0-bin
```
</p></details>

GrahamcOfBorg commented:
<!--REQUEST_ID=b6152b0d-76c6-4f32-a66d-47e1c1a59891-->
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.46747&attempt_id=b16ee4ad-deec-4b00-958e-6d2f58649b66)

Attempted: ipfs-migrator

<details><summary>Partial log (click to expand)</summary><p>

```
/nix/store/fr1wks0qh0811pvi3as5a7zg4yvcn0jw-ipfs-migrator-1.4.0-bin
```
</p></details>

GrahamcOfBorg commented:
<!--REQUEST_ID=b6152b0d-76c6-4f32-a66d-47e1c1a59891-->
Failure on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.46747&attempt_id=8f4d72c4-516d-4a8f-b257-c10a61750ee8)

Attempted: ipfs-migrator

<details><summary>Partial log (click to expand)</summary><p>

```
github.com/ipfs/fs-repo-migrations/ipfs-2-to-3/Godeps/_workspace/src/github.com/whyrusleeping/go-logging
github.com/ipfs/fs-repo-migrations/Godeps/_workspace/src/github.com/anacrolix/utp
github.com/ipfs/fs-repo-migrations/Godeps/_workspace/src/github.com/jbenet/go-multihash
github.com/ipfs/fs-repo-migrations/Godeps/_workspace/src/github.com/jbenet/go-multiaddr
github.com/ipfs/fs-repo-migrations/Godeps/_workspace/src/github.com/jbenet/go-multiaddr-net/utp
github.com/ipfs/fs-repo-migrations/Godeps/_workspace/src/github.com/jbenet/go-multiaddr-net
github.com/ipfs/fs-repo-migrations/sharness/dependencies/pollEndpoint
github.com/ipfs/fs-repo-migrations/stump
builder for '/nix/store/3v433gb5pqhslnn7pp9w1w8bdkkkncg1-ipfs-migrator-1.4.0.drv' failed with exit code 3
error: build of '/nix/store/3v433gb5pqhslnn7pp9w1w8bdkkkncg1-ipfs-migrator-1.4.0.drv' failed
```
</p></details>

GrahamcOfBorg commented:
<!--REQUEST_ID=275d5fb8-12a5-49fd-a33d-3d239cbc7392-->
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.46747&attempt_id=4ca21cf8-34ef-4f09-8a93-ad06efd17724)

Attempted: ipfs-migrator

<details><summary>Partial log (click to expand)</summary><p>

```
stripping (with command strip and flags -S) in /nix/store/j8pkyd66wabxcppnr47ha55vrb78mhz1-ipfs-migrator-7-bin/bin
patching script interpreter paths in /nix/store/j8pkyd66wabxcppnr47ha55vrb78mhz1-ipfs-migrator-7-bin
checking for references to /build in /nix/store/j8pkyd66wabxcppnr47ha55vrb78mhz1-ipfs-migrator-7-bin...
cannot find section .dynamic
cannot find section .dynamic
cannot find section .dynamic
cannot find section .dynamic
cannot find section .dynamic
strip is /nix/store/h0lbngpv6ln56hjj59i6l77vxq25flbz-binutils-2.30/bin/strip
/nix/store/j8pkyd66wabxcppnr47ha55vrb78mhz1-ipfs-migrator-7-bin
```
</p></details>

GrahamcOfBorg commented:
<!--REQUEST_ID=275d5fb8-12a5-49fd-a33d-3d239cbc7392-->
Success on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.46747&attempt_id=c102eeb2-05bf-4caa-a48c-980314685796)

Attempted: ipfs-migrator

<details><summary>Partial log (click to expand)</summary><p>

```
stripping (with command strip and flags -S) in /nix/store/blpgpms02rifmjmyjwv4vqk5hi4rb8cx-ipfs-migrator-7-bin/bin
patching script interpreter paths in /nix/store/blpgpms02rifmjmyjwv4vqk5hi4rb8cx-ipfs-migrator-7-bin
checking for references to /build in /nix/store/blpgpms02rifmjmyjwv4vqk5hi4rb8cx-ipfs-migrator-7-bin...
cannot find section .dynamic
cannot find section .dynamic
cannot find section .dynamic
cannot find section .dynamic
cannot find section .dynamic
strip is /nix/store/y4ymnvgxygpq05h03kyzbj572zmh6zla-binutils-2.30/bin/strip
/nix/store/blpgpms02rifmjmyjwv4vqk5hi4rb8cx-ipfs-migrator-7-bin
```
</p></details>

GrahamcOfBorg commented:
<!--REQUEST_ID=275d5fb8-12a5-49fd-a33d-3d239cbc7392-->
Success on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.46747&attempt_id=a11965f7-314e-4ba1-aebc-305f83aa8e58)

Attempted: ipfs-migrator

<details><summary>Partial log (click to expand)</summary><p>

```
find: '/nix/store/4idqmkxfmfwgq0jc6mw17sjmvyyqmzx0-ipfs-migrator-7': No such file or directory
find: '/nix/store/4idqmkxfmfwgq0jc6mw17sjmvyyqmzx0-ipfs-migrator-7': No such file or directory
find: '/nix/store/4idqmkxfmfwgq0jc6mw17sjmvyyqmzx0-ipfs-migrator-7': No such file or directory
find: '/nix/store/4idqmkxfmfwgq0jc6mw17sjmvyyqmzx0-ipfs-migrator-7': No such file or directory
find: '/nix/store/4idqmkxfmfwgq0jc6mw17sjmvyyqmzx0-ipfs-migrator-7': No such file or directory
find: '/nix/store/4idqmkxfmfwgq0jc6mw17sjmvyyqmzx0-ipfs-migrator-7': No such file or directory
find: '/nix/store/4idqmkxfmfwgq0jc6mw17sjmvyyqmzx0-ipfs-migrator-7': No such file or directory
find: '/nix/store/4idqmkxfmfwgq0jc6mw17sjmvyyqmzx0-ipfs-migrator-7': No such file or directory
find: '/nix/store/4idqmkxfmfwgq0jc6mw17sjmvyyqmzx0-ipfs-migrator-7': No such file or directory
/nix/store/9bnnz5csj1gdajrmlvsb0c51fmigwjpc-ipfs-migrator-7-bin
```
</p></details>

xeji committed:
ipfs-migrator: update dependencies in deps.nix

Most dependencies are vendored in the upstream source anyway,
deps.nix just contains the few that are not.
Dependencies were updated manually because go2nix doesn't
work correctly for this package - it produces an empty deps.nix.

GrahamcOfBorg commented:
<!--REQUEST_ID=1ba40530-621f-4c9a-b848-d36f33556e13-->
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.46747&attempt_id=111e9c1b-9a37-4b68-b195-e07bcef62534)

Attempted: ipfs-migrator

<details><summary>Partial log (click to expand)</summary><p>

```
stripping (with command strip and flags -S) in /nix/store/yrn922crbnvk0kgkm478wfcm2vpsx8my-ipfs-migrator-7-bin/bin
patching script interpreter paths in /nix/store/yrn922crbnvk0kgkm478wfcm2vpsx8my-ipfs-migrator-7-bin
checking for references to /build in /nix/store/yrn922crbnvk0kgkm478wfcm2vpsx8my-ipfs-migrator-7-bin...
cannot find section .dynamic
cannot find section .dynamic
cannot find section .dynamic
cannot find section .dynamic
cannot find section .dynamic
strip is /nix/store/h0lbngpv6ln56hjj59i6l77vxq25flbz-binutils-2.30/bin/strip
/nix/store/yrn922crbnvk0kgkm478wfcm2vpsx8my-ipfs-migrator-7-bin
```
</p></details>

GrahamcOfBorg commented:
<!--REQUEST_ID=1ba40530-621f-4c9a-b848-d36f33556e13-->
Success on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.46747&attempt_id=c4f53026-1240-41cf-b410-abb90a2093a5)

Attempted: ipfs-migrator

<details><summary>Partial log (click to expand)</summary><p>

```
stripping (with command strip and flags -S) in /nix/store/40rmhfplg71jhrjw1frh1s0gq571agzn-ipfs-migrator-7-bin/bin
patching script interpreter paths in /nix/store/40rmhfplg71jhrjw1frh1s0gq571agzn-ipfs-migrator-7-bin
checking for references to /build in /nix/store/40rmhfplg71jhrjw1frh1s0gq571agzn-ipfs-migrator-7-bin...
cannot find section .dynamic
cannot find section .dynamic
cannot find section .dynamic
cannot find section .dynamic
cannot find section .dynamic
strip is /nix/store/y4ymnvgxygpq05h03kyzbj572zmh6zla-binutils-2.30/bin/strip
/nix/store/40rmhfplg71jhrjw1frh1s0gq571agzn-ipfs-migrator-7-bin
```
</p></details>

GrahamcOfBorg commented:
<!--REQUEST_ID=1ba40530-621f-4c9a-b848-d36f33556e13-->
Success on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.46747&attempt_id=7bffca92-0b41-43a9-a903-eb3e714f8864)

Attempted: ipfs-migrator

<details><summary>Partial log (click to expand)</summary><p>

```
find: '/nix/store/wdz9z3f9h22z90dxszdwmkkwcjq7bzxk-ipfs-migrator-7': No such file or directory
find: '/nix/store/wdz9z3f9h22z90dxszdwmkkwcjq7bzxk-ipfs-migrator-7': No such file or directory
find: '/nix/store/wdz9z3f9h22z90dxszdwmkkwcjq7bzxk-ipfs-migrator-7': No such file or directory
find: '/nix/store/wdz9z3f9h22z90dxszdwmkkwcjq7bzxk-ipfs-migrator-7': No such file or directory
find: '/nix/store/wdz9z3f9h22z90dxszdwmkkwcjq7bzxk-ipfs-migrator-7': No such file or directory
find: '/nix/store/wdz9z3f9h22z90dxszdwmkkwcjq7bzxk-ipfs-migrator-7': No such file or directory
find: '/nix/store/wdz9z3f9h22z90dxszdwmkkwcjq7bzxk-ipfs-migrator-7': No such file or directory
find: '/nix/store/wdz9z3f9h22z90dxszdwmkkwcjq7bzxk-ipfs-migrator-7': No such file or directory
find: '/nix/store/wdz9z3f9h22z90dxszdwmkkwcjq7bzxk-ipfs-migrator-7': No such file or directory
/nix/store/8b7an4izkr6bzhf03bm30cn9cij14gjp-ipfs-migrator-7-bin
```
</p></details>

xeji commented:
cc @elitak @fpletz

xeji commented:
backported: 95b11f3cb9d..347723980a1
